### PR TITLE
Pipeline2.0: add API to access sink/src components linked to a buffer

### PR DESCRIPTION
**Pull request by marcinszkudlinski**
step 5 - make all components use API to access sink/src components linked to buffers

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

File: src/include/sof/audio/component.h [hunk 1]
```diff
@@ -46,13 +46,14 @@ struct dai_ts_data;
...
-#define COMP_STATE_ACTIVE	5	/**< Component active */
-#define COMP_STATE_PRE_ACTIVE	6	/**< Component after early initialisation */
+#define COMP_STATE_NOT_EXIST    0	/**< Component does not exist */
+#define COMP_STATE_INIT		1	/**< Component being initialised */
+#define COMP_STATE_READY	2	/**< Component inactive, but ready */
+#define COMP_STATE_SUSPEND	3	/**< Component suspended */
+#define COMP_STATE_PREPARE	4	/**< Component prepared */
+#define COMP_STATE_PAUSED	5	/**< Component paused */
+#define COMP_STATE_ACTIVE	6	/**< Component active */
+#define COMP_STATE_PRE_ACTIVE	7	/**< Component after early initialisation */
```
Comment (kv2019i) [inline src/include/sof/audio/component.h:56]: @marcinszkudlinski  Any strong reason to reassign the numbers and not add NOT_EXIST at the end? I guess one possibility is some comparisons on state in existing code. I guess this is ok, but yeah, some of us have been staring at the raw component state numbers for years in log files and will need to relearn these new numbers, so not a small change.
UPDATE: there are quite a few "if state > READY" checks in existing code, has to be done like this.

File: src/audio/buffers/comp_buffer.c [hunk 2]
```diff
@@ -480,9 +480,10 @@ void comp_update_buffer_produce(struct comp_buffer *buffer, uint32_t bytes)
 	/* return if no bytes */
 	if (!bytes) {
 #if CONFIG_SOF_LOG_DBG_BUFFER
+		struct comp_dev *src_component = comp_buffer_get_source_component(buffer);
```
Comment (kv2019i) [inline src/audio/buffers/comp_buffer.c:483]: This would need a "__unused" attribute to not create compiler warnings when buf_dbg() is a no-op.

**Review by kv2019i — APPROVED** [review]
If this needs an update, the "__unused" attributes can be added, but not a blocker. Otherwise looks good.

The component state renumbering will surprise some developers looking at logs, but there's probably no way to keep the current schema (and it's not THAT big a deal in the end).

File: src/include/sof/audio/component.h (hunk 1 above)
Comment (marcinszkudlinski) [reply src/include/sof/audio/component.h:56]: > UPDATE: there are quite a few "if state > READY" checks in existing code, has to be done like this.

yes, that was the reason

**Pull request merged**